kafkastats/src/main/java/com/pinterest/doctorkafka/util/OperatorUtil.java

```diff
@@ -87,12 +101,100 @@ public static boolean pingKafkaBroker(String host, int port, int timeout) {
       socket.connect(new InetSocketAddress(host, port), timeout);
       return true;
     } catch (UnknownHostException e) {
+      LOG.warn("Ping failure, host: " + host, e);
       return false;
     } catch (IOException e) {
+      LOG.warn("Ping failure IO, host: " + host, e);
       return false; // Either timeout or unreachable or failed DNS lookup.
     }
   }
 
+  public static boolean fetchData(String host, int port, String topic, int partition) {
+    LOG.info("Fetching data from host {}, topic {}, partition {}", host, topic, partition);
+    SimpleConsumer consumer = new SimpleConsumer(host, port,
```

Comment:
As SimpleConsumer will be deprecated soon, could we change to call the underneath api to get the data? 

https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java#L226

Reply:
The new consumer wouldn't let us to poll the followers.  Even though they might deprecate SimpleConsumer, the underlying FetchRequest would still remain.

Reply:
Could you use something similar to https://gist.github.com/yuyang08/656c59fcca9fe3e40bf01864f8f3f739 for fetching data, instead of using SimpleConsumer?  That will keep DoctorKafka from being impacted by SimpleConsumer removal.